sklearn/metrics/cluster/unsupervised.py

```diff
@@ -255,3 +258,58 @@ def calinski_harabaz_score(X, labels):
     return (1. if intra_disp == 0. else
             extra_disp * (n_samples - n_labels) /
             (intra_disp * (n_labels - 1.)))
+
+
+def prediction_strength_score(labels_train, labels_test):
+    """Compute the prediction strength score.
+
+    For each test cluster, we compute the proportion of observation pairs
+    in that cluster that are also assigned to the same cluster by the
+    training set centroids. The prediction strength is the minimum of this
+    quantity over the k test clusters.
+
+    The best value is 1.0 (if the assignments of `labels_train` and
+    `labels_test` are identical) and the worst value is 0 (if all samples of
+    one cluster of `labels_test` are not co-members of some cluster in
+    `labels_train`).
+
+    Parameters
+    ----------
+    labels_train : array-like, shape (``n_test_samples``,)
+        Predicted labels for each sample in the the test data
+        based on clusters derived from independent training data.
+
+    labels_test : array-like, shape (``n_test_samples``,)
+        Predicted labels for each sample in the test data
+        based on clusters derived from the same data.
+
+    Returns
+    -------
+    score : float
+        The resulting prediction strength score.
+
+    References
+    ----------
+    .. [1] `Robert Tibshirani and Guenther Walther (2005). "Cluster Validation
+    by Prediction Strength". Journal of Computational and Graphical Statistics,
+    14(3), 511-528. <http://doi.org/10.1198/106186005X59243>_`
+    """
+    check_consistent_length(labels_train, labels_test)
+
+    labels_train = check_array(labels_train, dtype=np.int32, ensure_2d=False,
+                               warn_on_dtype=True)
+    labels_test = check_array(labels_test, dtype=np.int32, ensure_2d=False,
+                              warn_on_dtype=True)
+
+    n_clusters = max(np.unique(labels_train).shape[0],
+                     np.unique(labels_test).shape[0])
+    if n_clusters == 1:
+        return 1.0  # by definition
+
+    C = contingency_matrix(labels_train, labels_test)
```

Comment:
ideally we would use `sparse=True` as with the other metrics. In which case, `*` needs to be replaced by `.multiply` and `C - 1` needs to be replaced by something like `_nonzero_add(C, -1)` with:
```
def _nonzero_add(M, d):
    out = M.copy()
    out.data += d
    return out
```